lib/email/sender.rb

```diff
@@ -226,6 +226,23 @@ def send
       end
 
       MessageBuilder.custom_headers(SiteSetting.email_custom_headers).each do |key, _|
+        # Any custom headers added via MessageBuilder that are doubled up here
+        # with values that we determine should be set to the last value, which is
+        # the one we determined. Our header values should always override the email_custom_headers.
+        #
+        # While it is valid via RFC5322 to have more than one value for certain headers,
+        # we just want to keep it to one, especially in cases where the custom value
+        # would conflict with our own.
+        #
+        # See https://datatracker.ietf.org/doc/html/rfc5322#section-3.6 and
+        # https://www.rubydoc.info/github/mikel/mail/Mail/Header
+        custom_header = @message.header[key]
+        if custom_header.is_a?(Array)
+          our_value = custom_header.last.value
+          @message.header[key] = nil
+          @message.header[key] = our_value
```

Comment:
```suggestion
          @message.header[key] = our_value
```

I'm curious about the reason why we need to set the key to `nil` first.

Reply:
Heh I knew someone would ask :) It's because if you do `@message.header[key] = our_value` without setting it to nil first it just adds another value onto the array!

Reply:
Ooo so it isn't a hash we're dealing with here. Got it :+1: